src/components/course/data/hooks.jsx

```diff
@@ -720,24 +733,46 @@ export const useUserSubsidyApplicableToCourse = ({
         courseDetails,
         firstEnrollablePaidSeatPrice: courseService?.activeCourseRun?.firstEnrollablePaidSeatPrice,
       });
-      const subsidy = getSubsidyToApplyForCourse({
+      return getSubsidyToApplyForCourse({
         applicableSubscriptionLicense: licenseApplicableToCourse,
         applicableCouponCode: findCouponCodeForCourse(couponCodes, catalogsWithCourse),
         applicableEnterpriseOffer: findEnterpriseOfferForCourse({
           enterpriseOffers: canEnrollWithEnterpriseOffers ? enterpriseOffers : [],
-          catalogList: catalogsWithCourse,
+          catalogsWithCourse,
           coursePrice,
         }),
       });
-      if (!hasEMETRedeemability) {
-        // If course cannot be redeemed through a subsidy access policy, set the applicable
-        // pre-EMET subsidy as `userSubsidyApplicableToCourse`. This conditional avoids
-        // overwriting `userSubsidyApplicableToCourse` when it has already been set.
-        setUserSubsidyApplicableToCourse(subsidy);
-      }
-      setLegacyUserSubsidyApplicableToCourse(subsidy);
     };
-    retrieveApplicableSubsidy();
+
+    // If have not yet determined whether there is an applicable subsidy access policy, fallback
+    // to checking against legacy subsidies.
+    if (!applicableUserSubsidy) {
+      retrieveApplicableLegacySubsidy().then((legacyUserSubsidyApplicableToCourse) => {
+        setUserSubsidyApplicableToCourse(legacyUserSubsidyApplicableToCourse);
+      });
+    }
+
+    if (applicableUserSubsidy) {
+      setUserSubsidyApplicableToCourse(applicableUserSubsidy);
+    } else if (missingSubsidyAccessPolicyReason) {
+      setMissingUserSubsidyReason({
+        reason: missingSubsidyAccessPolicyReason.reason,
+        userMessage: missingSubsidyAccessPolicyReason.userMessage,
+        actions: getMissingSubsidyReasonActions({
+          reasonType: missingSubsidyAccessPolicyReason.reason,
+          enterpriseAdminUsers,
+        }),
+      });
+    } else {
+      setMissingUserSubsidyReason({
+        reason: noApplicableUserSubsidyReasonType,
```

Comment:
Is noApplicableUserSubsidyReasonType set anywhere?

Reply:
Nope, good catch. Updating.